### ISSUE TITLE
Ability to use another pager than less

### ISSUE THREAD

**Issue by ccoVeille** [bensadeh/tailspin#128]
Hi,

I'm interested in being able to use another pager than the default one, which is `less` currently.

@TornaxO7 suggested minus in #95 and worked on an implementation with #100 https://github.com/arijit79/minus

I saw the debate about being able to use minus with `--minus`

I would be more interested in using ov https://github.com/noborus/ov made by @noborus

Other people may have other preference, after all, that's the magic with open source.

So my request for enhancement would be about being able to use another pager.

a common way to configure it in tools is to do it via environment variable.

Like:
- `BAT_PAGER` for https://github.com/sharkdp/bat
- `DELTA_PAGER` for https://github.com/dandavison/delta

I don't mind how to be able to configure it. It could be in the `config.toml` file, but I would like to be able to configure the pager I'm using.

I'm curious about your thoughts

Thanks

**Comment by ccoVeille** [comment]
I understand the possible debate to be able to use rust only code, but it would limit yourself to pager available in rust.

**Comment by bensadeh** [comment]
Hi @ccoVeille and thanks for opening this ticket!

To begin with the conclusion: I'm happy to add support for other pagers.

Some context: as the **default** pager, it makes sense for me to use `less` as it is so ubiquitous. That being said, `tailspin` is really just a log file highlighter, and the tool you choose to use to view the log files doesn't really matter to me (or to `tailspin`). I specifically built it in a way to decouple the presentation (paging) functionality from the highlighting functionality for this very purpose.

I can't give a specific timeline for when this will be implemented, but in theory it should be quite simple.

On a side note: I'd still like to see support for `minus` as a pager option (in addition to `less`), but it became difficult for me personally to implement. In the future, I want the use of pager in tailspin to be completely configurable, however I don't see us moving away from `less` as the default pager.

**Comment by ccoVeille** [comment]
Thanks for replying in such detailed way. 

I'm aligned with what you said.

I understand we will have to wait. A enhancement request is like a item to Christmas list. Sometimes you get it, sometimes it takes time, sometimes you never have it.

Thanks for your work on tailspin

**Comment by HRKings** [comment]
Thanks for tailspin! 
The ability to use any pager would be great! I currently use [ov](https://github.com/noborus/ov) which seems very fast and handles tailing very nicely.

Take your time with on the development and thanks again!

**Comment by gcflymoto** [comment]
Would a hack which replaces `less` with a wrapper script that instead executed `ov` @ccoVeille? I had to replace `less` options with `ov` options.

`--ignore-case --RAW-CONTROL-CHARS`

**Comment by ccoVeille** [comment]
> Would a hack which replaces `less` with a wrapper script that instead executed `ov` @ccoVeille? I had to replace `less` options with `ov` options.
> 
> `--ignore-case --RAW-CONTROL-CHARS`

I'm sorry @gcflymoto but I don't get what you meant. Mainly, because I feel like some words are missing.

Especially here:

> Would a hack which replaces `less` with a wrapper script that instead executed `ov` @ccoVeille?

I'll try to reply anyway.

For me TAILSPIN_PAGER or any variable that would be available would allow to do this

```env
TAILSPIN_PAGER="ov --ignore-case --RAW-CONTROL-CHARS"
```

**Comment by gcflymoto** [comment]
Tailspin calls the pager with different options in different modes. So the tailspin author would need to provide a mechanism to achieve the flexibility of different pagers. My hack replaces the command "less" with a script that modifies the passed in less options to ones that are suitable for "ov". I have updated my original comment

**Comment by ccoVeille** [comment]
> Tailspin calls the pager with different options in different modes. 

Thanks for this, it's clearer now.

So I assume you created a shell script that has precedence over less command via the order of folder in your PATH variable maybe.

I understand this as an hack. Not, that I don't find it good. But it's not a solution I would recommend for everyone.

I'm happy you found something that fits your need.

But I would prefer if tailspain could find a way to overload the pager, and now I have understood it depends on the mode, I understand it wouldn't be as simple as I thought

**Comment by gcflymoto** [comment]
Yes exactly. It's a temporary hack that shouldn't be used as a final means.

**Comment by bensadeh** [comment]
Hi everyone,

I've added preliminary support for using custom pagers in c4a52d2. It uses the `TAILSPIN_PAGER` env variable to run a provided pager.

I've tested it with ov locally and it seems to work, but I'll be happy if you can test it out and see if there are any issues with it before we release it.

I am currently testing it with this command:

```console
TAILSPIN_PAGER="ov -f [FILE]" tspin example-logs/example1
```

Note that the `[FILE]` string is mandatory and is just a placeholder for where the filename will be interpolated by tailspin later on.

I will add more thorough documentation later, just wanted to share with you here first.

**Comment by bensadeh** [comment]
Released in `5.0.0`. Hope it works well for you, let me know if there are any issues.
